spec/asyncapi.md

```diff
@@ -1159,9 +1181,11 @@ Describes a message received on a given channel and operation.
 Field Name | Type | Description
 ---|:---:|---
 <a name="messageObjectMessageId"></a>messageId | `string` | Unique string used to identify the message. The id MUST be unique among all messages described in the API. The messageId value is **case-sensitive**. Tools and libraries MAY use the messageId to uniquely identify a message, therefore, it is RECOMMENDED to follow common programming naming conventions.
+<a name="messageObjectType"></a>type | `string` | The type of the message. Should be set if message is used in a [Operation](#operationsObject) using request/reply pattern. Allowed values are: `generic`, `request`, `response`. Default value is `generic`.
```

Comment:
No it dont. @derberg wanted this field at our discussion in spring to give the code generator a hint.
I see here no benefits and would remove it if every one agree.

Reply:
I dont see a reason for this either 🙂 Tbh it creates somewhat a confusion I feel like 🤔

Reply:
@derberg  Any response from your side?

Reply:
I do not need to know if message is a request or reply.
I need a way to say that there will be a reply to this message, and what is it.

so I need to specify in `ping` message, that in reply I will get `pong` message. I do not need `pong` message to specify that it is a reply as this is what the operation is for, it will tell me that `pong` is only something I can consume, but not produce

does it make sense?

but looking at 3.0 I think it is no longer needed because of how operations work now 🤔

Reply:
so for websocket use case `replyChannel` do not help much, as there is one channel anyway

🤔 

and I know with `replyChannel` you are coming from this world -> https://www.enterpriseintegrationpatterns.com/patterns/messaging/RequestReply.html

but do we have to call it `replyChannel`? why not `reply`? as we anway do not specify the channel there, but actually, where info about the channel can be found. This way, it can be easily extended in the future if needed.

Reply:
@derberg 

> but looking at 3.0 I think it is no longer needed because of how operations work now 🤔

but now you cannot define on which message (I mean request message) on a given channel you will perform "reply action". You can only define messages at channel level and you cannot specify if a given operation (connected to a given channel) perform logic only on e.g. `ping` message and then reply `ping` message to another channel - https://github.com/asyncapi/spec/blob/next-major-spec/spec/asyncapi.md#operationObject

Reply:
Ok i will remove it

Reply:
@derberg at `operation` level it is called `reply` to be future save.
At level `message` it is called `replyChannel` because her you refer to specific header or payload field containing a string for the channel. Channel means in this case the topic/destination not the channel object from schema.